tests/integration/test_notebooks_gpu.py

```diff
@@ -1,7 +1,7 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
-import shutil
+from tempfile import TemporaryDirectory
```

Comment:
Interestingly, when I do `pytest` with `-s` to print something to console it failed if `TemporaryDiretory` or `tempdir` is used; `gettempdir()` works.... 😂

Reply:
does this work for you?
```
with TemporaryDirectory() as tmp_dir:

```

Reply:
lol ... w/o '-s' does this work?

Reply:
Neither of the above worked for me... `pytest` fixture `tmpdir` works, so this is so far the best practice.

Reply:
@yueguoguo Can you double check what was the reason for the failure? @miguelgfierro and I had discussed and we want to use `with TemporaryDirectory() as td:` back again, because:

pytest's fixture (tmpdir or tmp_path or tmp_factory) keeps 3-latest temp folders and we don't want that (also saw someone complaining about the old folders are not removed completely from their CI machine).

I tested using TemporaryDirectory from my side and both unit and smoke tests are passed, with `-s`

Reply:
Tested and it works with the context manager. Thanks!

Reply:
Let me refactor all tempdir things in a new issue and PR! I will merge the current PR as soon as possible since the main fix is the bugfix.